torchao/prototype/parq/quant/config_torchao.py

```diff
@@ -193,8 +193,9 @@ def _attach_hf_quantization_config(
         if not hasattr(module, "weight"):
             continue
 
+        # Do not quantize pointers to tied weights or normalization layers
         data_ptr = module.weight.data_ptr()
-        if data_ptr in seen_data_ptrs:  # do not re-quantize tied weight
+        if data_ptr in seen_data_ptrs or name.endswith("norm"):
```

Comment:
maybe it's better to filter out the class instead? Is there a common class we can use here for that?

Reply:
"norm" in name might be more general I think

Reply:
also using `data_ptr` seems a bit fragile and hacky I feel, any other way to detect sharing? (like checking id of weight?)

Reply:
@jerryzh168 Could you share more about checking id of the weight? I've used `data_ptr` consistently in the PARQ code and [this](https://dev-discuss.pytorch.org/t/weight-sharing-on-cuda/701/2) suggests it's a valid way to check that the underlying storage is the same (correct me if I'm wrong)

Reply:
Sure, I added a check for membership in `model._tied_weights_keys` instead of `seen_data_ptrs`. I also added a test case for what happens in `PreTrainedModel.from_pretrained` when given a quantization config. That method iterates over `model.state_dict().items()` instead of `model.named_parameters()`, which includes tied weights.

Reply:
I see. I think model._tied_weights_keys will give us the names of the pointers. I'm not sure what method remove_duplicates is using to distinguish pointer params from the reference